### PR TITLE
Adding robust fitting to the 2D fitting function, 

### PR DESCRIPTION
**Pull request by cophus**
Robust fitting can now be used in .fit origin like so:
py4DSTEM.process.calibration.fit_origin(qx0_meas,qy0_meas,fitfunction='plane',robust=True)

Additionally I have added a 2x2 Bezier fitting function which can be used with this fitfunction:
py4DSTEM.process.calibration.fit_origin(qx0_meas,qy0_meas,fitfunction='bezier_two',robust=True)

I cleaned fit_2D( ... ) to have less conditional statements and perform all fits with the same line of code.

And finally this pull request also includes readme updates / typo fixes.

### CODE REVIEW

File: py4DSTEM/process/calibration/origin.py [hunk 1]
```diff
@@ -182,6 +183,7 @@ def fit_origin(qx0_meas,qy0_meas,mask=None,fitfunction='plane',returnfitp=False)
         mask                    (2b boolean array) ignore points where mask=True
         fitfunction             (str) must be 'plane' or 'parabola'
         returnfitp              (bool) if True, returns the fit parameters
+        robust					(bool) if True, perform robust fitting
```
Comment (bsavitzky) [inline py4DSTEM/process/calibration/origin.py:186]: Add doc line for robust_steps and robust_thresh.  Make 'robust' line align with above lines.

File: py4DSTEM/process/fit/fit.py [hunk 2]
```diff
@@ -24,20 +25,27 @@ def fit_1D_gaussian(xdata,ydata,xmin,xmax):
     A,mu,sigma = scale*popt[0],popt[1],popt[2]
     return A,mu,sigma
 
-def fit_2D(function, data, data_mask=None, popt_guess=None):
+def fit_2D(function, data, data_mask=None, popt_guess=None, 
+    robust=False, robust_steps=3, robust_thresh=2):
     """
     Performs a 2D fit, where the fit function takes its first input in the form of a length 2
     vector (ndarray) of (x,y) positions, followed by the remaining parameters, and the data to
-    fit takes the form of an (n,m) shaped array.
```
Comment (bsavitzky) [inline py4DSTEM/process/fit/fit.py:31]: I think maybe this line was deleted by accident?

File: py4DSTEM/process/fit/fit.py [hunk 3]
```diff
@@ -46,27 +54,39 @@ def fit_2D(function, data, data_mask=None, popt_guess=None):
                     same shape as data, containing the fit values
     """
     shape = data.shape
+    shape1D = [1,np.prod(shape)]
     x,y = np.arange(shape[0]),np.arange(shape[1])
+    x,y = np.linspace(0, 1, shape[0]),np.linspace(0, 1, shape[1])
```
Comment (bsavitzky) [inline py4DSTEM/process/fit/fit.py:59]: add a comment at the end of this line that x,y are normalized coordinates

File: py4DSTEM/process/fit/fit.py [hunk 4]
```diff
@@ -46,27 +54,39 @@ def fit_2D(function, data, data_mask=None, popt_guess=None):
...
-            popt, pcov = curve_fit(function, xy_known, data_1D, p0=popt_guess)
+    # if robust fitting is turned off, set number of robust iterations to 0
+    if robust==False:
+        robust_steps=0
+
+    # least squares fitting - 1st iteration 
+    for k in range(robust_steps+1):
+        if k == 0:
+            if popt_guess is None:
+                popt = np.zeros((1,len(signature(function).parameters)-1))
```
Comment (bsavitzky) [inline py4DSTEM/process/fit/fit.py:73]: add a line with else: popt_guess = popt, OR fix the naming so there's one variable

**Review by bsavitzky — CHANGES_REQUESTED** [review]
Looks great!  See comments below, there was one bug, hopefully can now be squashed.

File: py4DSTEM/process/calibration/origin.py (hunk 1 above)
Comment (cophus) [reply py4DSTEM/process/calibration/origin.py:186]: done

File: py4DSTEM/process/fit/fit.py (hunk 2 above)
Comment (cophus) [reply py4DSTEM/process/fit/fit.py:31]: Yes it was!  Fixed.

File: py4DSTEM/process/fit/fit.py (hunk 3 above)
Comment (cophus) [reply py4DSTEM/process/fit/fit.py:59]: comment added!

File: py4DSTEM/process/fit/fit.py (hunk 4 above)
Comment (cophus) [reply py4DSTEM/process/fit/fit.py:73]: popt_guess has been renamed to just popt everywhere.

**Comment by cophus** [comment]
Ready for new review

**Pull request merged**